stdlib/public/TensorFlow/Gradients.swift

```diff
@@ -355,10 +355,19 @@ func _adjointRsqrt<T : Differentiable & FloatingPoint>(
   return -seed / 2 * pow(originalValue, 3)
 }
 
-func _adjointSquared<T : Differentiable & FloatingPoint>(
+@inlinable
+func _adjointLogSoftmax<T : Differentiable & FloatingPoint>(
   _ seed: Tensor<T>, _ originalValue: Tensor<T>, _ x: Tensor<T>
 ) -> Tensor<T> {
-  return 2 * x * seed
+  let seed = seed.broadcast(like: originalValue)
```

Comment:
We should no longer broadcast seed to the original value in any derivative.

Reply:
I see - instead, we should assume that seed has same shape as original result and require users to specify non-scalar initial seeds via pullbacks?

That seems mathematically correct, tests using `gradient` that assume the scalar seed `Tensor(1)` gets broadcasted need to be rewritten using `pullback`.

Reply:
Correct!